### PR TITLE
(fix) adding sanity check to catch empty default channel

### PR DESCRIPTION
**Pull request by gallettilance**
**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/registry/populator.go [hunk 1]
```diff
@@ -94,6 +95,15 @@ func (i *DirectoryPopulator) globalSanityCheck(imagesToAdd []*ImageInput) error
...
 				break
 			}
+			if channel == image.annotationsFile.GetDefaultChannelName() {
+				containsDefault = true
+			}
+		}
+		if !containsDefault {
+			_, err := i.querier.GetBundleForChannel(context.TODO(), image.bundle.Package, image.annotationsFile.GetDefaultChannelName())
+			if err != nil {
+				errs = append(errs, EmptyDefaultChannel{ErrorString: "Cannot set an empty default channel"})
```
Comment (njhale) [inline pkg/registry/populator.go:105]: If this error string isn't changing, shouldn't it be hard-coded in the `Error() string` method?

File: pkg/registry/populator.go [hunk 2]
```diff
@@ -94,6 +95,15 @@ func (i *DirectoryPopulator) globalSanityCheck(imagesToAdd []*ImageInput) error
 				errs = append(errs, PackageVersionAlreadyAddedErr{ErrorString: "Bundle already added that provides package and csv"})
 				break
 			}
+			if channel == image.annotationsFile.GetDefaultChannelName() {
+				containsDefault = true
+			}
+		}
+		if !containsDefault {
```
Comment (njhale) [inline pkg/registry/populator.go:102]: historically, we've inferred the default channel when |channels|=1 -- is this happening at some other level (e.g. when getting the annotations file)? did we drop this behavior at some point?

Comment (gallettilance) [reply pkg/registry/populator.go:102]: yep that still happens today and is what is returned by the `GetDefaultChannel` function

File: pkg/registry/populator.go (hunk 1 above)
Comment (gallettilance) [reply pkg/registry/populator.go:105]: I forgot to add `err` to the string - my bad

**Comment by hasbro17** [comment]
@gallettilance Just checking in, since this PR has been sitting for a while now. Do you still have time to rebase and address the review? If not, I can take over.

**Comment by gallettilance** [comment]
@hasbro17 Please take over if this is still needed

**Comment by ankitathomas** [comment]
https://github.com/operator-framework/operator-registry/commit/809651742864f61db6099163eb1b2539d6d92cd3#diff-5dd7ba96a38b29d88c95d131248cc0dc6bf9b3b62adbe50612975076d1d3b96d

With the semver based channel head/default channel selection, we can close this out

**Pull request closed without merging**